lib/ansible/module_utils/redfish_utils.py

```diff
@@ -455,6 +474,87 @@ def get_disk_inventory(self, systems_uri):
     def get_multi_disk_inventory(self):
         return self.aggregate(self.get_disk_inventory)
 
+    def get_volume_inventory(self, systems_uri):
+        result = {'entries': []}
+        controller_list = []
+        volume_list = []
+        volume_results = []
+        # Get these entries, but does not fail if not found
+        properties = ['Id', 'Name', 'RAIDType', 'VolumeType', 'BlockSizeBytes',
+                      'Capacity', 'CapacityBytes', 'CapacitySources',
+                      'Encrypted', 'EncryptionTypes', 'Identifiers',
+                      'Operations', 'OptimumIOSizeBytes', 'AccessCapabilities',
+                      'AllocatedPools', 'Status']
+
+        # Find Storage service
+        response = self.get_request(self.root_uri + systems_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        if 'SimpleStorage' not in data and 'Storage' not in data:
+            return {'ret': False, 'msg': "SimpleStorage and Storage resource \
+                     not found"}
+
+        if 'Storage' in data:
+            # Get a list of all storage controllers and build respective URIs
+            storage_uri = data[u'Storage'][u'@odata.id']
+            response = self.get_request(self.root_uri + storage_uri)
+            if response['ret'] is False:
+                return response
+            result['ret'] = True
+            data = response['data']
+
+            if data[u'Members']:
```

Comment:
Avoids a `KeyError: u'Members'` exception:

```suggestion
            if data.get('Members'):
```

Reply:
Many thanks, billdodd, I fixed the code as you suggested.